### ISSUE TITLE
install-spark doesn't extract number of instances properly

### ISSUE THREAD

**Issue by derrickburns** [awslabs/emr-bootstrap-actions#43]
```
[hadoop@ip-10-83-134-205 ~]$ grep /mnt/var/lib/info/job-flow.json -e "CORE" -A 4 | grep -e requestedInstanceCount | cut -d':' -f2 | sed  's/\s\+//g'
12,
```

Notice the trailing comma

**Comment by christopherbozeman** [comment]
Part of issue https://github.com/awslabs/emr-bootstrap-actions/issues/38.

**Comment by christopherbozeman** [comment]
Derrick... is this still an issue?

**Comment by derrickburns** [comment]
I stopped using install-spark -x, but I will try again.

**Comment by derrickburns** [comment]
I tested your code, abc while it appears to set the Spark parameters correctly, I am NOT getting all the executors that are allocated. This is consistent behavior.

**Comment by christopherbozeman** [comment]
@derrickburns - Can I get a cluster id and application id to reference demonstrating the issue?   Also, is there sufficient memory to fulfill all the executors requested when considering the amount of driver memory being requested?

**Comment by derrickburns** [comment]
Try this one: j-8YQLFX9G9PV5.  There is only one application.

It appears to me that there are 11 non-driver executors running and 1
driver executor.  However, I requested 13 core and 1 master.  So I am
confused.

**Comment by derrickburns** [comment]
Also, the -x option only works in yarn-client mode.

**Comment by derrickburns** [comment]
How do you specify the memory of the Application Master separate from the
memory for the driver when running in yarn-client mode?

**Comment by christopherbozeman** [comment]
In regards to specifying memory of the AM separate from driver, see spark.yarn.am.memory which is available in Spark 1.3.
